envi/cli.py

```diff
@@ -391,30 +382,38 @@ def do_config(self, line):
 
         if len(args) == 1:
             parts = args[0].split('=', 1)
-            subname, optname = parts[0].split('.', 1)
+            subnames = parts[0].split('.')
 
-            subcfg = self.config.getSubConfig(subname, add=False)
-            if subcfg is None:
-                self.vprint('No Such Config Section: %s' % subname)
-                return
+            cfg = self.config
+            for subname in subnames[:-1]:
+                cfg = cfg.getSubConfig(subname, add=False)
+                if cfg is None:
+                    self.vprint('No Such Config Section: %s' % subname)
+                    return
 
-            optval = subcfg.get(optname)
+            optname = subnames[-1]
+            optval = cfg.get(optname)
             if optval is None:
                 self.vprint('No Such Config Option: %s' % optname)
                 return
 
             if len(parts) == 2:
-                newval = json.loads(parts[1])
```

Comment:
this json approach is problematic.  it requires quotes around string values, otherwise it errors out.
unfortunately the quotes we hand in get eaten by shlex previously.  therefore, we need double-quotes or escaped quotes, which is obtuse.
so i ripped json.loads() out and just parsed it based on the existing value (since we don't allow setting of new options here)